packages/nightingale-track-canvas/src/nightingale-track-canvas.ts

```diff
@@ -0,0 +1,316 @@
+import { createEvent, customElementOnce } from "@nightingale-elements/nightingale-new-core";
+import NightingaleTrack, { Feature, FeatureLocation, Shapes } from "@nightingale-elements/nightingale-track";
+import { BaseType, select, Selection } from "d3";
+import { html } from "lit";
+import { drawRange, drawSymbol, drawUnknown, shapeCategory } from "./utils/draw-shapes";
+import { last, RangeCollection, Refresher } from "./utils/utils";
+
+
+type Fragment = FeatureLocation["fragments"][number]
+type ExtendedFragment = Fragment & { featureIndex: number, location: FeatureLocation };
```

Comment:
Is `location: FeatureLocation` needed?

Reply:
You're right, i don't use it anywhere. I'll remove it.